pbm/backup/backup.go

```diff
@@ -122,16 +128,98 @@ func (b *Backup) Run(bcp pbm.BackupCmd) (err error) {
 		if err != nil {
 			return errors.Wrap(err, "unable to finish cluster backup")
 		}
+		err = b.oplog(oplog, oplogTS, stg, rsMeta.OplogName, bcp.Compression)
+		if err != nil {
+			return errors.Wrap(err, "oplog")
+		}
 		return b.dumpClusterMeta(bcp.Name, stg)
 	case pbm.ReplRoleShard:
-		return b.waitForCluster(bcp.Name)
+		err := b.waitForCluster(bcp.Name)
+		if err != nil {
+			return errors.Wrap(err, "waiting for cluster")
+		}
+		err = b.oplog(oplog, oplogTS, stg, rsMeta.OplogName, bcp.Compression)
+		return errors.Wrap(err, "oplog")
 	case pbm.ReplRoleUnknown:
 		return errors.Wrap(err, "unknow node role in sharded cluster")
 	}
 
 	return nil
 }
 
+const maxReplicationLagTimeSec = 21
+
+// NodeSuits checks if node can perform backup
+func NodeSuits(bcp pbm.BackupCmd, node *pbm.Node) (bool, error) {
+	im, err := node.GetIsMaster()
+	if err != nil {
+		return false, errors.Wrap(err, "get isMaster data for node")
+	}
+
+	if im.IsMaster && !bcp.FromMaster {
+		return false, nil
+	}
+
+	status, err := node.Status()
+	if err != nil {
+		return false, errors.Wrap(err, "get node status")
+	}
+
+	replLag, err := node.ReplicationLag()
+	if err != nil {
+		return false, errors.Wrap(err, "get node replication lag")
+	}
+
+	return replLag < maxReplicationLagTimeSec && status.Health == pbm.NodeHealthUp &&
+			(status.State == pbm.NodeStatePrimary || status.State == pbm.NodeStateSecondary),
+		nil
+}
+
+// RWErr multierror for the read/compress/write-to-store operations set
+type RWErr struct {
```

Comment:
Do we need to expose this error?
If yes may we use one of the options:
- Expose pointer instead of value. This is more useful in errors.Cause(err).(struct) comparison.  
- Create IsRWErr func where we can hide logic

Reply:
Actually, yes, no need to make this error type exported now